tools/lkl/include/lkl.h

```diff
@@ -19,6 +19,10 @@ extern "C" {
 #define strtok_r strtok_s
 #endif
 
+#if defined(__ANDROID__) && __LKL__BITS_PER_LONG == 32
```

Comment:
Hmm, I wonder if we need this even for non Android platforms. This would explain #360.

Reply:
i don't have other 32bits environment than arm32 so, not sure if this is true or not.
do you see any issue on mingw32 ?

Reply:
That is the strange thing, I don't see any issue on either 32 or 64bit in the boot.c test. I thought that there might be some issues with libhijack but I disassemble it and the right number (0x19) is used there as well.

However, I noticed that we have a bug in asm-generic/unistd.h

```
#if (__LKL__BITS_PER_LONG == 32 || defined(__SYSCALL_COMPAT)) && \
     defined(__ARCH_WANT_SYSCALL_OFF_T)
...
#define __lkl__NR_fcntl 1052
```

Note that __ARCH_WANT_SYSCALL_OFF_T and __SYSCALL_COMPAT are not prefixed with LKL, which in certain build environments may be defined and set __lkl__NR_fcntl to an invalid value.

Can you confirm that this is the issue for Android?

Reply:
Here is an quick band-aid patch that should fix the issue. With it applied, we should not need the above.

```
diff --git a/arch/lkl/scripts/headers_install.py b/arch/lkl/scripts/headers_install.py
index 6365da9..7b4ab22 100755
--- a/arch/lkl/scripts/headers_install.py
+++ b/arch/lkl/scripts/headers_install.py
@@ -165,6 +165,8 @@ p = re.compile("typedef\s+(struct|union)\s+\w*\s*{[^\\{\}]*}\W*(\w+)\s*;", re.M|
 find_ml_symbols(p, defines)
 defines.add("siginfo_t")
 defines.add("sigevent_t")
+defines.add("__ARCH_WANT_SYSCALL_OFF_T")
+defines.add("__SYSCALL_COMPAT")
 p = re.compile("struct\s+(\w+)\s*\{")
 find_symbols(p, structs)
 structs.add("iovec")
```